monai/data/box_utils.py

```diff
@@ -0,0 +1,291 @@
+    dst_mode: Union[str, BoxMode, Type[BoxMode], None] = None,
+) -> NdarrayOrTensor:
+    """
+    This function converts the boxes in src_mode to the dst_mode
+    Args:
+        boxes: source bounding box, Nx4 or Nx6 torch tensor or ndarray
+        src_mode: source box mode. If mode is not given, this func will assume mode is StandardMode.
+        dst_mode: target box mode. If mode is not given, this func will assume mode is StandardMode.
+        src_mode and dst_mode can be:
+            #. str: choose from monai.utils.enums.BoundingBoxMode, including
+                "xyxy": boxes has format [xmin, ymin, xmax, ymax]
+                "xyzxyz": boxes has format [xmin, ymin, zmin, xmax, ymax, zmax]
+                "xxyy": boxes has format [xmin, xmax, ymin, ymax]
+                "xxyyzz": boxes has format [xmin, xmax, ymin, ymax, zmin, zmax]
+                "xyxyzz": boxes has format [xmin, ymin, xmax, ymax, zmin, zmax]
+                "xywh": boxes has format [xmin, ymin, xsize, ysize]
+                "xyzwhd": boxes has format [xmin, ymin, zmin, xsize, ysize, zsize]
+                "ccwh": boxes has format [xcenter, ycenter, xsize, ysize]
+                "cccwhd": boxes has format [xcenter, ycenter, zcenter, xsize, ysize, zsize]
+            #. BoxMode class: choose from
+                CornerCornerModeTypeA: equivalent to "xyxy" or "xyzxyz"
+                CornerCornerModeTypeB: equivalent to "xxyy" or "xxyyzz"
+                CornerCornerModeTypeC: equivalent to "xyxy" or "xyxyzz"
+                CornerSizeMode: equivalent to "xywh" or "xyzwhd"
+                CenterSizeMode: equivalent to "ccwh" or "cccwhd"
+            #. BoxMode instance
+            #. None: will assume mode is StandardMode
+    Returns:
+        boxes_dst: bounding box with target mode, does not share memory with original boxes
+
+    Example:
+        boxes = torch.ones(10,6)
+        box_convert_mode(boxes=boxes, src_mode="xyzxyz", dst_mode=CenterSizeMode)
+    """
+
+    # if mode not changed, return original box
+    src_boxmode = get_boxmode(src_mode)
+    dst_boxmode = get_boxmode(dst_mode)
+    if type(src_boxmode) is type(dst_boxmode):
+        return deepcopy(boxes)
+    # convert mode
+    else:
+        # convert numpy to tensor if needed
+        boxes_t, *_ = convert_data_type(boxes, torch.Tensor)
+
+        # convert boxes to corners
+        corners = src_boxmode.boxes_to_corners(boxes_t)
+
+        # check validity of corners
+        if not check_corners(corners):
+            raise ValueError("Given boxes has invalid values. The box size must be non-negative.")
+
+        # convert corners to boxes
+        boxes_t_dst = dst_boxmode.corners_to_boxes(corners)
+
+        # convert tensor back to numpy if needed
+        boxes_dst, *_ = convert_to_dst_type(src=boxes_t_dst, dst=boxes)
+        return boxes_dst
+
+
+def convert_box_to_standard_mode(
+    boxes: NdarrayOrTensor, mode: Union[str, BoxMode, Type[BoxMode], None] = None
```

Comment:
Here we must try to keep `boxes` and `mode` in-sync, that's why I proposed to define a `class` for the bounding box with mode property. I am OK for the current logic, let's keep it in mind for the future enhancement.

Thanks.

Reply:
Yes, I understand. We could discuss it in the future